backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/serialization/UniqSymbolName.kt

```diff
@@ -0,0 +1,72 @@
+package org.jetbrains.kotlin.backend.konan.serialization
+
+import org.jetbrains.kotlin.backend.konan.irasdescriptors.fqNameSafe
+import org.jetbrains.kotlin.backend.konan.irasdescriptors.name
+import org.jetbrains.kotlin.backend.konan.llvm.extensionReceiverNamePart
+import org.jetbrains.kotlin.backend.konan.llvm.functionName
+import org.jetbrains.kotlin.backend.konan.llvm.symbolName
+import org.jetbrains.kotlin.backend.konan.llvm.typeInfoSymbolName
+import org.jetbrains.kotlin.ir.declarations.*
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.name.Name
+
+// This is a little extension over what's used in real mangling
+// since some declarations never appear in the bitcode symbols.
+
+internal fun IrDeclaration.uniqSymbolName(): String = when (this) {
+    is IrFunction
+    -> this.uniqFunctionName
+    is IrProperty
+    -> this.symbolName
+    is IrClass
+    -> this.typeInfoSymbolName
+    is IrField
+    -> this.symbolName
+    is IrEnumEntry
+    -> this.symbolName
+    else -> error("Unexpected exported declaration: $this")
+}
+
+private val IrDeclarationParent.fqNameUnique: FqName
+    get() = when(this) {
+        is IrPackageFragment -> this.fqName
+        is IrDeclaration -> this.parent.fqNameUnique.child(this.uniqName)
+        else -> error(this)
+    }
+
+private val IrDeclaration.uniqName: Name
+    get() = when (this) {
+        is IrSimpleFunction -> Name.special("<${this.uniqFunctionName}>")
```

Comment:
So `IrSimpleFunction.fqNameUnique` seems to include parent's fqName twice. Is this intended?

Reply:
No-o, afaict  (:-z